src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { Button, Label, Select, TextInput } from 'flowbite-react';
+import {
+  FuelSourceUnit,
+  FuelSourceType,
+  UnregisteredFuelRequest,
+} from '../../schema/fuel.schema';
+
+interface FuelFormProps {
+  handleSubmit: (fuelData: UnregisteredFuelRequest) => void;
+}
+
+export default function FuelForm({ handleSubmit }: FuelFormProps) {
+  /* Initializing state */
+  const [fuelData, setfuelData] = useState({
+    fuel_source_type: 'ng' as FuelSourceType,
+    fuel_source_unit: 'btu' as FuelSourceUnit,
+    fuel_source_value: 0 as number,
+  });
+
+  function submitForm(e: React.FormEvent) {
+    e.preventDefault();
+    console.log(fuelData);
+    handleSubmit(fuelData);
+  }
+
+  /* Function that updates the state */
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    /* Destructuring the event.target object */
+    const { name, value, valueAsNumber } = event?.target;
+    setfuelData((prevFuelData) => {
+      return {
+        ...prevFuelData,
+        [name]: name == 'fuel_source_value' ? valueAsNumber : value,
```

Comment:
Since you lose `valueAsNumber`, do this: `[name]: name === 'fuel_source_value' ? parseFloat(value) : value,`

Reply:
Can you explain the difference?

Reply:
parseFloat takes a string and returns a number, in this case a number with a decimal. `value` is always a string.

Reply:
I think `valueAsNumber` only exists on the `React.ChangeEvent<HTMLInputElement>` type but not on `React.ChangeEvent<HTMLSelectElement>` type so if we use the union for the event `event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>` this won't work.

Reply:
Also, `==` is a loose equality, it generally is disfavored and `===` is preferred. `===` is a strict equality and is equal in function to other language's `==` function.

Reply:
I meant the difference between valueAsNumber and parseFloat. valueAsNumber is a HTML5 way to receive an input value as number.